.github/workflows/preview.yml

```diff
@@ -0,0 +1,99 @@
+name: Preview PR
+
+on:
+  workflow_run:
+    workflows: ["Build and publish"]
+    types: 
+      - completed
+
+jobs:
+# This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      
+      # Extracts PR from Head SHA
+      - name: Find PR
```

Comment:
did you consider following https://github.com/actions/checkout/issues/58#issuecomment-545769838 ?

Reply:
Unfortunately that only works if the event trigger is a PR itself, but for security and permission reasons, the trigger here is indirect (workflow run) instead.